bravado/async_http_client.py

```diff
@@ -44,116 +33,26 @@ def start_request(self, request_params):
-        deferred = agent.request(**request_params)
-
-        def response_callback(response):
-            """Callback for response received from server, even 4XX, 5XX possible
-            response param stores the headers and status code.
-            It needs a callback method to be registered to store the response
-            body which is provided using deliverBody
-            """
-            response.deliverBody(_HTTPBodyFetcher(request_params,
-                                                  response, finished_resp))
-        deferred.addCallback(response_callback)
-
-        def response_errback(reason):
-            """Error callback method like server not reachable or conn. refused
+        if isinstance(url, unicode):
+            url = url.encode('utf-8')
 
-            :param reason: The reason why request failed
-            :type reason: str
-            """
-            finished_resp.errback(reason)
-        deferred.addErrback(response_errback)
-
-        return finished_resp
-
-
-class AsyncResponse(object):
-    """
-    Remove the property text and content and make them as overridable attrs
-    """
-
-    def __init__(self, req, resp, data):
-        self.request = req
-        self.status_code = resp.code
-        self.headers = dict(resp.headers.getAllRawHeaders())
-        self.text = data
-
-    def raise_for_status(self):
-        """Raises stored `HTTPError`, if one occured.
-        """
-
-        http_error_msg = ''
-
-        if 400 <= self.status_code < 500:
-            http_error_msg = '%s Client Error' % self.status_code
-
-        elif 500 <= self.status_code < 600:
-            http_error_msg = '%s Server Error' % self.status_code
-
-        if http_error_msg:
-            raise HTTPError(http_error_msg, response=self)
-
-    def json(self, **kwargs):
-        return json.loads(self.text, **kwargs)
-
-
-class _HTTPBodyFetcher(Protocol):
-    """Class to receive callbacks from Twisted whenever
-    response is available.
-
-    Eventually AsyncResponse() is created on receiving complete response
-    """
-
-    def __init__(self, request, response, finished):
-        self.buffer = StringIO()
-        self.request = request
-        self.response = response
-        self.finished = finished
-
-    def dataReceived(self, data):
-        self.buffer.write(data)
-
-    def connectionLost(self, reason):
-        # Accepting PotentialDataLoss for servers with HTTP1.0
-        # and not sending Content-Length in the header
-        if reason.check(twisted.web.client.ResponseDone) or \
-                reason.check(twisted.web.http.PotentialDataLoss):
-            self.finished.callback(AsyncResponse(
-                self.request, self.response, self.buffer.getvalue()))
-        else:
-            self.finished.errback(reason)
+        return fido.fetch(url, **request_params)
 
 
 def stringify_body(request_params):
     """Wraps the data using twisted FileBodyProducer
     """
+    data = None
```

Comment:
Instead of adding `data = None` can we change each of these to just return immediately?

```python

if ...:
    return create_multipart_content(...)

if headers.get(...):
    return ..

return param_stringify_body(...)

```